cmds/wifi/wifi.go

```diff
@@ -38,36 +38,62 @@ func init() {
 	}
 }
 
+func generateConfig(essid string, id string, pass string) (conf []byte, err error) {
+	switch {
+	case essid != "" && id != "" && pass != "":
+		conf = []byte(fmt.Sprintf(eap, essid, id, pass))
+	case essid != "" && id == "" && pass != "":
+		conf, err = passphrase.Run(essid, pass)
+		if err != nil {
+			return nil, fmt.Errorf("essid: %v, pass: %v : %v", essid, pass, err)
+		}
+	case essid != "" && id == "" && pass == "":
+		conf = []byte(fmt.Sprintf(nopassphrase, essid))
+	default:
+		return nil, fmt.Errorf("Invalid Argument: essid: %v, id: %v, pass: %v", essid, id, pass)
+	}
+	return
+}
+
 func main() {
+	go startServer()
+
+	// Service
 	var (
 		iface = flag.String("i", "wlan0", "interface to use")
-		essid string
 		conf  []byte
+		err   error
 	)
 
 	flag.Parse()
 	a := flag.Args()
 
 	switch {
 	case len(a) == 3:
-		essid = a[0]
-		conf = []byte(fmt.Sprintf(eap, essid, a[2], a[1]))
+		conf, err = generateConfig(a[0], a[2], a[1])
```

Comment:
it seems a little awkward to me to duplicate the logic of generateConfig here and in the function too.

you might just call generateConfig with
generateConfig(a...)

and then in the function itself, case on the len of args.

see https://play.golang.org/p/mWdAHqBBKjh

Reply:
I agree with what you said. My purpose for doing this initially was to promote code reuse becuase I noticed that after we get inputs from client, we still need to go through the same logic to generate the conf. I think I have a fix for this awkwardness coming up in the next commit.